### DISCUSSION TITLE
How to run static analysis only on one recipe at a time?

### DISCUSSION THREAD

**Discussion by RaGrCobham** [priv-kweihmann/meta-sca#9137] [Q&A]
I have it enabled for multiple recipes, but it's nice to be able to run "bitbake recipe -c do_sca_pylint" and have it output only for that  instead of everything.

**Comment by priv-kweihmann** [comment]
The easiest way is to add 'SCA_ENABLED_MODULES = "pylint"' to the recipe and do `bitbake <recipe>`.

Currently SCA implementation consists of several tasks that are performed in a certain order.

Alternatively you could run

`bitbake recipe -c do_sca_pylint_core_report` which would run `do_sca_pylint` + `do_sca_pylint_core_report` and create internal representations (usually called `<module>.dm`, so in this case here `pylint.dm`) in the temp subdir of the recipe's workspace.
These are json that can be easily inspected.

**Reply by RaGrCobham** [reply]
I have the enabled modules inside a class, but when I inherit it in an image recipe, it doesn't propagate to run to the dependent recipes.

**Reply by RaGrCobham** [reply]
Basically, I want to be able to run SCA for an image and its dependencies that have it enabled AND be able to run only for a single one of them.

**Comment by RaGrCobham** [comment]
Ok, so apparently it's only listing which modules are using SCA but it doesn't mean that they'll be analyzed during that run.

**Reply by priv-kweihmann** [reply]
Yes - the variable just controls which SCA module could be run in a recipe (normal + images) - it doesn't mean that is actually runs the checks.
This is only determined by bitbake and its task dependency tree.

So what you would like to do is
- in an image recipe run some checks
- inherit the results of all packages that are part of this particular image and combine them into a result package

is that correct?

**Reply by RaGrCobham** [reply]
I got it partially working. I have a class with all static-analysis settings including 'inherit sca'. Nothing is inherited globally.

I inherited that class into the recipes I want, including the image depending on those recipes 

Now I can analyze the recipes individually (although as a side effect it analyzes the dependencies of those individual files as well!) and images and its dependencies as a whole.

**Reply by priv-kweihmann** [reply]
That doesn't sound right...
The tasks should just run in the recipe that `sca` is inherited to - but not the dependencies

**Reply by RaGrCobham** [reply]
Perhaps there's confusion. By dependencies I mean only those recipes that DO inherit sca, not all of them.
